### PR TITLE
[CELEBORN-1250] Fix license issues for release 0.4.0

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
To fix license issues.

### Why are the changes needed?
https://lists.apache.org/thread/sto05pgd213ymr6z1csvmkhbg2dcrwls

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

**Comment by pan3793** [comment]
why target branch is 0.4?

**Comment by FMX** [comment]
> why target branch is 0.4?

@pan3793 Because release 0.4 has encountered license issues. And I want to make sure that 0.4 can be released soon.

**Comment by pan3793** [comment]
I think the main have the same issue, why not fix in main and backport?

**Comment by FMX** [comment]
I think cherry-picking this PR to the main would be a wise choice. This license issue exists in the main branch, and the main branch might have some other license issue. I think we'll need to check the license and notice for every release.

File: LICENSE-binary [hunk 1]
```diff
@@ -256,7 +257,6 @@ org.apache.hadoop:hadoop-mapreduce-client-jobclient
 org.apache.hadoop:hadoop-mapreduce-client-shuffle
 org.apache.hadoop:hadoop-yarn-api
```
Comment (pan3793) [inline LICENSE-binary:258]: I think the diff of this patch makes sense, but for existing items, are those hadoop libs included in our binary tarball, except for `hadoop-client-api` and `hadoop-client-runtime`?

Comment (FMX) [reply LICENSE-binary:258]: I think other Hadoop libs can be removed.

**Comment by FMX** [comment]
After running UTs on local machines, I think the UT failures are not relevant to this PR.

**Comment by pan3793** [comment]
@FMX https://github.com/apache/incubator-celeborn/pull/2253#issuecomment-1907971633

**Comment by waitinfuture** [comment]
gentle ping @camper42 PTAL

**Comment by FMX** [comment]
> @FMX [#2253 (comment)](https://github.com/apache/incubator-celeborn/pull/2253#issuecomment-1907971633)

I'll make a follow-up to fix UTs.

**Comment by waitinfuture** [comment]
> LGTM, I think all items listed in IMPC voting thread are addressed by this PR

This PR checks dependency differences agains v0.2.0-incubating.

**Comment by waitinfuture** [comment]
> I think cherry-picking this PR to the main would be a wise choice. This license issue exists in the main branch, and the main branch might have some other license issue. I think we'll need to check the license and notice for every release.

Yes, and we should check LICENSE whenever dependency changes (including version upgrading)

File: LICENSE-binary [hunk 2]
```diff
@@ -285,7 +274,7 @@ com.fasterxml.jackson.core:jackson-databind
 com.fasterxml.jackson.core:jackson-annotations
 com.fasterxml.jackson.core:jackson-core
 com.thoughtworks.paranamer:paranamer
```
[the remark this answers is not in the document]
Comment (FMX) [reply LICENSE-binary:276]: OK, updated.

**Review by carp84 — APPROVED** [review]
+1.

Good catch on the additional issues of LICENSE- and NOTICE-binary files. I checked the RC4 binary distribution and confirmed the updated LICENSE-binary file could reflect the exact bundled dependencies (no more no less).

**Comment by cfmcgrady** [comment]
Thanks all!

merging to branch-0.4(v0.4.0).

I will initiate the vote for the next RC later today.

**Comment by camper42** [comment]
Nice!

**Pull request closed without merging**